### PR TITLE
Plugins: Core plugins register via backend factory provider

### PR DESCRIPTION
**Pull request by wbrowne**
**What this PR does / why we need it**:
Instead of Core plugins registering themselves directly, use a registry to track these instead which is responsible for serving their backend implementations during the normal plugin loading flow

### CODE REVIEW

**Review by marefr — APPROVED** [review]
LGTM, added a comment

**Review by sunker — APPROVED** [review]
What a great improvement. Thanks for this!

File: pkg/plugins/backendplugin/provider/provider.go [hunk 1]
```diff
@@ -43,6 +48,17 @@ var RendererProvider PluginBackendProvider = func(_ context.Context, p *plugins.
 }
 
 var DefaultProvider PluginBackendProvider = func(_ context.Context, p *plugins.Plugin) backendplugin.PluginFactoryFunc {
+	// TODO check for executable
```
[the remark this answers is not in the document]
Comment (wbrowne) [reply pkg/plugins/backendplugin/provider/provider.go:51]: Yeah... this is something I'd like to do now, but unfortunately mangles a bunch of tests. Something I want to revisit ASAP

[the remark this answers is not in the document]
Comment (marefr) [reply pkg/plugins/backendplugin/provider/provider.go:51]: Sure

**Review by marefr — APPROVED** [review]
LGTM

**Pull request merged**